pkg/server/endpoints/endpoints.go

```diff
@@ -311,33 +310,32 @@ func (e *Endpoints) getTLSConfig(ctx context.Context) func(*tls.ClientHelloInfo)
 }
 
 // getCerts queries the datastore and returns a TLS serving certificate(s) plus
-// the current CA root bundle.
+// the current CA root bundle, alongside with federated bundles.
 func (e *Endpoints) getCerts(ctx context.Context) ([]tls.Certificate, *x509.CertPool, error) {
-	bundle, err := e.DataStore.FetchBundle(dscache.WithCache(ctx), e.TrustDomain.IDString())
+	listBundlesResponse, err := e.DataStore.ListBundles(ctx, &datastore.ListBundlesRequest{})
 	if err != nil {
-		return nil, nil, fmt.Errorf("get bundle from datastore: %w", err)
-	}
-	if bundle == nil {
-		return nil, nil, errors.New("bundle not found")
+		return nil, nil, fmt.Errorf("list bundle from datastore: %w", err)
 	}
+	caPool := x509.NewCertPool()
 
-	var caCerts []*x509.Certificate
-	for _, rootCA := range bundle.RootCas {
-		rootCACerts, err := x509.ParseCertificates(rootCA.DerBytes)
-		if err != nil {
-			return nil, nil, fmt.Errorf("parse bundle: %w", err)
+	for _, bundle := range listBundlesResponse.Bundles {
```

Comment:
We should probably scope which bundles we use here to the trust domain for the server and other trust domains expected in admin_ids.